internal/credentials/xds/handshake_info.go

```diff
@@ -85,40 +85,21 @@ func GetHandshakeInfo(attr *attributes.Attributes) *HandshakeInfo {
 //
 // Safe for concurrent access.
```

Comment:
Delete this now?  Or say it's immutable?

Reply:
I feel like technically it's fine to call any methods on it concurrently since it's all read only and set at init time, but since it's not really important to call out/doesn't apply I'll delete it.